### PR TITLE
Store VocabSheet items as JSON within VocabSheet instead of as separate table

### LINKED ISSUE #698: Store vocab sheet items inline in vocab_sheets table to stay within heroku DB plan row limits
Currently `VocabSheet` has many `Item` and during busy times the number of items stored can take us up to and over the 10k DB rows limit on our Heroku DB plan. Using a different plan isn't within the budget in the medium term so we should change items to be stored inline (possibly as a jsonb) in the VocabSheet model.

### PR DESCRIPTION
**Pull request by eoinkelly**
Change implementation of `Item` to not store as separate model.

* Start storing items within the `VocabSheet` as a JSON blob
* Convert `Item` from an `ActiveRecord` object to `ActiveModel` to preserve the public interface that the rest of the app expects.
* Add methods to `VocabSheet for handling Item storage and retrieval
* Remove unused `notes` field from Item

The motivation behind this change is staying under 10k DB rows on our current Heroku hosting plan because there is no extra funding for moving to a larger plan right now.

Minor bugs I fixed along the way:

* Fix JS error when changing name of vocab sheet
    * This didn't result in an error users would notice (the name change was saved and shown to them on the page) but it was causing an exception in the console
* Upgrade rubocop to 0.67 because old version could not parse Ruby 2.6 properly

This is a fairly substantial change so should get some manual testing in staging too @DSRU 

Closes #698

### CODE REVIEW

**Comment by eoinkelly** [comment]
Travis can't run the migration for some reason, will look into it later.

**Comment by Br3nda** [comment]
Add this to travis.yml

```
services:
  - postgresql
addons:
  postgresql: "10"
  apt:
    packages:
    - postgresql-10
    - postgresql-client-10
```

**Comment by Br3nda** [comment]
(but also check what version we have on heroku)

File: app/models/item.rb
Comment (eoinkelly) [reply app/models/item.rb:15]: I respectfully disagree here Mr. Bot.

**Comment by eoinkelly** [comment]
Heroku is running Postgres 9.6.14 (which definitely supports JSONB columns). I have updated Travis to also use 9.6

**Comment by Br3nda** [comment]
I'd like to give this a good test before merging. Can look on Friday

File: app/models/item.rb
Comment (Br3nda) [reply app/models/item.rb:15]: I've approved on codeclimate. Also the bot is a ms.

**Comment by eoinkelly** [comment]
Yup that seems like a good idea. I'm creating a review app on Heroku for it so we can all have a good click around

File: db/migrate/20190716173959_remove_items.rb
Comment (Br3nda) [inline db/migrate/20190716173959_remove_items.rb:2]: Shouldn't we migrate existing vocab sheets here too?

**Review by elspeth-rabid — COMMENTED** [review]
I am also reviewing this tomorrow @eoinkelly 📚

File: db/migrate/20190716173959_remove_items.rb
Comment (eoinkelly) [reply db/migrate/20190716173959_remove_items.rb:2]: I assumed we wouldn't bother with data migration given that vocab sheets are quite short lived but maybe that's being a bit 🤠 ?

Comment (Br3nda) [reply db/migrate/20190716173959_remove_items.rb:2]: i guess we don't know. We could do some analysis on the data to work out the usage patterns of people who use vocab sheets

**Review by Br3nda — APPROVED** [review]
works great

**Comment by eoinkelly** [comment]
Closes issue #698  (forgot to mention this in original description)

**Comment by Br3nda** [comment]
@eoinkelly merge this when you're ready.

File: app/models/vocab_sheet.rb
Comment (elspeth-rabid) [inline app/models/vocab_sheet.rb:10]: Are you a method documentation convert @eoinkelly ?

**Review by elspeth-rabid — APPROVED** [review]
Nice work and thanks @eoinkelly !

**Comment by Br3nda** [comment]
@eoinkelly anything left here? Shall we merge this?

File: app/models/vocab_sheet.rb
Comment (eoinkelly) [reply app/models/vocab_sheet.rb:10]: Kinda :-) I was trying to mimic the existing API that the controllers depended on so I needed to get the types straight in my head.

**Comment by eoinkelly** [comment]
Yes I reckon we should.

**Comment by eoinkelly** [comment]
I have a pretty full meetings day but I'll merge tonight when I have enough time to watch it go through.

**Comment by DSRU** [comment]
Just tested on staging (which is running very slow on Chrome, a bit better on Firefox). All seems to be working except for broken links to the images when saved to pdf directly. If saving as pdf via the print menu, images download correctly. 
vocab sheet on staging

**Pull request merged**
